pypit/artrace.py

```diff
@@ -3448,15 +3499,27 @@ def multislit_tilt(slf, msarc, det, maskval=-999999.9):
     ordcen = slf._pixcen[det - 1].copy()
     fitxy = [settings.argflag['trace']['slits']['tilts']['order'], 1]
 
+    # maskslit
+    if slf._maskslits[det-1] is not None:
+        mask = slf._maskslits[det-1] & (maskslit==1)
+    else:
+        mask = maskslit
+    slf._maskslits[det-1] = mask
+    gdslits = np.where(mask == 0)[0]
+
+    # Final tilts image
+    final_tilts = np.zeros_like(msarc)
+
     # Now trace the tilt for each slit
-    for o in range(arccen.shape[1]):
+    #for  o in range(arccen.shape[1]):
+    for oo, slit in enumerate(gdslits):
```

Comment:
Shouldn't we go over all slits, and then check that the slit is good? `arccen` might be a different shape to `gdslits`, right?  (EDIT: I see now your comment in the docstring of `get_censpec`. We should probably work towards this goal of padding with zeros or masking the slits that have no spectrum)

same is true for `lordloc` and `rordloc` below.

Reply:
Indeed, a mask has been implemented on the lines above.
Is that not sufficient?

Am adding a TODO in get_censpec.  Not sure it was worth
raising this to an Issue.